modules/tasks/tasks/src/main/java/org/motechproject/tasks/util/SchedulerTriggerRegistration.java

```diff
@@ -0,0 +1,82 @@
+package org.motechproject.tasks.util;
+
+import org.motechproject.tasks.contract.ActionEventRequest;
+import org.motechproject.tasks.contract.ChannelRequest;
+import org.motechproject.tasks.contract.EventParameterRequest;
+import org.motechproject.tasks.contract.TriggerEventRequest;
+import org.motechproject.tasks.ex.ValidationException;
+import org.motechproject.tasks.service.ChannelService;
+import org.osgi.framework.BundleContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import javax.annotation.PostConstruct;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class SchedulerTriggerRegistration {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SchedulerTriggerRegistration.class);
+
+    private BundleContext bundleContext;
+    private ChannelService channelService;
+
+
+    @Autowired
+    public SchedulerTriggerRegistration(BundleContext bundleContext, ChannelService channelService) {
+        this.bundleContext = bundleContext;
+        this.channelService = channelService;
+    }
+
+    @PostConstruct
+    public void updateTaskInfo() {
+        LOGGER.info("Updating tasks integration");
+
+        try {
+            updateChannel();
+        } catch (ValidationException e) {
+            LOGGER.error("Channel generated was not accepted by tasks due to validation errors", e);
```

Comment:
Kinda comical error message - tasks didn't accept the channel it generated by itself? Should we catch anything here?